meta/include/gnuradio-4.0/meta/UncertainValue.hpp

```diff
@@ -0,0 +1,331 @@
+    using ValueTypeU = UncertainValueType_t<U>;
+    if constexpr (UncertainValueLike<T> && UncertainValueLike<U>) {
+        using ResultType = decltype(lhs.value * rhs.value);
+        if constexpr (complex_like<ValueTypeT> || complex_like<ValueTypeU>) {
+            // we are dealing with complex numbers -> use standard uncorrelated calculation
+            ResultType newUncertainty;
+            if constexpr (std::is_arithmetic_v<ValueTypeT> && complex_like<ValueTypeU>) {
+                // LHS is real, RHS is complex
+                newUncertainty = { std::sqrt(std::pow(lhs.uncertainty / std::real(rhs.value), 2)), std::sqrt(std::pow(std::imag(rhs.uncertainty) * lhs.value / std::norm(rhs.value), 2)) };
+            } else if constexpr (complex_like<ValueTypeT> && std::is_arithmetic_v<ValueTypeU>) {
+                // LHS is complex, RHS is real
+                newUncertainty = { std::hypot(std::real(lhs.uncertainty) / rhs.value, rhs.uncertainty * std::real(lhs.value) / std::pow(rhs.value, 2)),
+                                   std::sqrt(std::pow(std::imag(lhs.uncertainty) / rhs.value, 2)) };
+            } else {
+                newUncertainty = { std::hypot(std::real(lhs.uncertainty) / std::real(rhs.value), std::real(rhs.uncertainty) * std::real(lhs.value) / std::norm(rhs.value)),
+                                   std::hypot(std::imag(lhs.uncertainty) / std::imag(rhs.value), std::imag(rhs.uncertainty) * std::imag(lhs.value) / std::norm(rhs.value)) };
+            }
+
+            return UncertainValue<ResultType>{ lhs.value / rhs.value, newUncertainty };
+        } else {
+            // both ValueType[T,U] are arithmetic uncertainties
+            auto combinedUncertainty = std::hypot(lhs.uncertainty / rhs.value, rhs.uncertainty * lhs.value / std::pow(rhs.value, 2));
+            return UncertainValue<ResultType>{ lhs.value / rhs.value, combinedUncertainty };
+        }
+    } else if constexpr (UncertainValueLike<T> && arithmetic_or_complex_like<ValueTypeU>) {
+        return T{ lhs.value / rhs, lhs.uncertainty / std::abs(rhs) };
+    } else if constexpr (arithmetic_or_complex_like<ValueTypeT> && UncertainValueLike<U>) {
+        auto rhsMagSquared = std::norm(rhs.value);
+        return U{ lhs / rhs.value, rhs.uncertainty * std::abs(lhs) / rhsMagSquared };
+    } else {
+        static_assert(std::is_arithmetic_v<ValueTypeT> && std::is_arithmetic_v<ValueTypeU>);
+        return lhs / rhs; // unlikely to be called due to default '/' definition
+    }
+}
+
+template<UncertainValueLike T, typename U>
+inline constexpr T &
+operator/=(T &lhs, const U &rhs) noexcept {
+    lhs = lhs / rhs;
+    return lhs;
+}
+
+} // namespace gr::meta
+
+namespace std { // std:: basic math overloads
+
+template<typename T>
+    requires std::is_floating_point_v<T>
+[[nodiscard]] inline constexpr T
+hypot(const std::complex<T> &a, const std::complex<T> &b) noexcept { // doesn't exist in the STL for some reason
```

Comment:
This is even more dangerous. All exceptions to the 'don''t define things in std::' require at least one user type. And here, all types are std ones. (so, even if the general comment from above is something that we could ignore, this isn't)